codegen/build-oas.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+set -eux -o pipefail
+
+version=$1 # e.g. 2024-07
+modules=("control" "data")
+
+destination="src/pinecone-generated-ts-fetch"
+build_dir="build"
+
+update_apis_repo() {
+	echo "Updating apis repo"
+	pushd codegen/apis
+		git fetch
+		git pull
+		just build
+	popd
+}
+
+verify_spec_version() {
+	local version=$1
+	echo "Verifying spec version $version exists in apis repo"
+	if [ -z "$version" ]; then
+		echo "Version is required"
+		exit 1
+	fi
+
+	verify_directory_exists "codegen/apis/_build/${version}"
+}
+
+verify_file_exists() {
+	local filename=$1
+	if [ ! -f "$filename" ]; then
+		echo "File does not exist at $filename"
+		exit 1
+	fi
+}
+
+verify_directory_exists() {
+	local directory=$1
+	if [ ! -d "$directory" ]; then
+		echo "Directory does not exist at $directory"
+		exit 1
+	fi
```

Comment:
Tried checking out your branch and running locally, but it seems like there are issues with the submodule and verifying if the directory exists. Not sure if related to the above, or just the fact that I'm trying to do this in a branch:

```
+ echo 'Verifying spec version 2024-07 exists in apis repo'
Verifying spec version 2024-07 exists in apis repo
+ '[' -z 2024-07 ']'
+ verify_directory_exists codegen/apis/_build/2024-07
+ local directory=codegen/apis/_build/2024-07
+ '[' '!' -d codegen/apis/_build/2024-07 ']'
+ echo 'Directory does not exist at codegen/apis/_build/2024-07'
Directory does not exist at codegen/apis/_build/2024-07
+ exit 1
```